test/views/SubmitFormViewHiddenTextSpec.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views
+
+import java.time.LocalDate
+import forms.SubmitVatReturnForm
+import models.VatObligation
+import models.nrs.{CY, EN}
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import play.api.i18n.Lang
+
+class SubmitFormViewHiddenTextSpec extends ViewBaseSpec {
+
+  "When the submit form view is rendered " when {
+
+    val obligation: VatObligation = VatObligation(LocalDate.parse("2019-01-12"), LocalDate.parse("2019-04-12"), LocalDate.parse("2019-05-12"), "18AA")
+
+    "Rendering the submit_form page in welsh" should {
+      val languageOption: Lang = Lang.apply(CY.languageCode)
+
+      "have the language set as welsh" in {
+        languageOption.language shouldBe "cy"
+      }
+
+      lazy val view = views.html.submit_form(
+        "18AA",
+        Some("ABC Studios"),
+        flatRateScheme = true,
+        obligation,
+        SubmitVatReturnForm.nineBoxForm,
+        isAgent = false
+      )(fakeRequest, welshMessages, mockAppConfig, user, languageOption)
+      lazy implicit val document: Document = Jsoup.parse(view.body)
+
+      "have the welsh messages file" in {
+        welshMessages.lang.toString shouldBe "Lang(cy)"
+      }
+
+      "have the correct hidden label for box 1" in {
+        elementText("label[for=box1]") shouldBe "Swm Blwch 1 TAW a godwyd gennych ar werthiannau a chyflenwadau eraill"
+      }
+
+      "have the correct hidden label for box 2" in {
```

Comment:
Checking the content for one box should be sufficient. It doesn't make sense to test for 2 out of 9 boxes vs 1 out of 9 boxes, especially if the logic isn't different